homeassistant/components/ccm15/config_flow.py

```diff
@@ -0,0 +1,87 @@
+"""Config flow for Midea ccm15 AC Controller integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+import homeassistant.helpers.config_validation as cv
+
+from .climate import CCM15Coordinator
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Optional(CONF_PORT, default=80): cv.port,
+    }
+)
+
+
+async def validate_input(
+    hass: HomeAssistant, data: dict[str, Any], existing_titles: list[str]
+) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+
+    host: str = data[CONF_HOST]
+    for title in existing_titles:
+        if title == host:
+            raise DuplicateEntry
+
+    hub = CCM15Coordinator(hass, host, data[CONF_PORT])
+    if not await hub.async_test_connection():
+        raise CannotConnect
+
+    # Return info that you want to store in the config entry.
+    return {
+        CONF_HOST: data[CONF_HOST],
+        CONF_PORT: data[CONF_PORT],
+    }
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Midea ccm15 AC Controller."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            try:
+                existing_titles = [
+                    entry.title for entry in self._async_current_entries()
```

Comment:
Entry. Title is user changeable so no solid way to check uniqueness.

Is there some kind of serial we can fetch? Other wise use self.async abort entries match.

Reply:
No, the only real unique id is the IP address of the ccm15 module. The integration does not even provide a MacId.

Reply:
Then still, checking for an entry title is wrong.